lockfile/src/javascript.rs

```diff
@@ -80,9 +80,12 @@ impl Parse for PackageLock {
                 let resolved = get_field(keys, "resolved")
                     .ok_or_else(|| anyhow!("Dependency '{name}' is missing \"resolved\" key"))?;
 
+                // Handle aliased dependencies.
+                let name = get_field(keys, "name").unwrap_or_else(|| name.into());
```

Comment:
Will the entry even be valid if there is no `name` field? I wouldn't imagine that npm is parsing the name out of the key as a fallback, but who knows 🤷

Reply:
Yeah you can check the test file for this, having no `name` field is actually the "normal" case. Only for aliases the name field is even populated.

It's very unfortunate but otherwise I wouldn't even do all this ridiculous path splitting.